File: projekt_2/gra/resources/Zwierzeta/Owca.java
package gra.resources.Zwierzeta;

import gra.resources.backend.*;

public class Owca extends Zwierze {
    private final static int OWCA_SILA = 4;
    private final static int OWCA_INIC = 4;
    private final static char OWCA_CHAR = 'O';
    private final static int OWCA_POLOG = 5;
    private final static int OWCA_PELNOLETNOSC = 10;
    private final static String OWCA_NAZWA = "Owca";

    public Owca(Swiat swiat, int x, int y) { super(swiat, x, y, OWCA_SILA, OWCA_INIC, OWCA_PELNOLETNOSC); }

    public boolean Koliduj(Organizm somsiad, boolean czyAtakuje) {
        return false;
    }
    public void dodajPolog() { this.polog = OWCA_POLOG; }
    public Organizm dajPotomka(Wsp n_wsp) {
        Organizm tmp = new Owca(swiat, n_wsp.x, n_wsp.y);
        swiat.DodajOrganizm(tmp);
        return tmp;
    }
    public boolean czyTenSamGatunek(Organizm o) { if (o instanceof Owca) { return true; } else { return false; } }
    public char getChar() { return OWCA_CHAR; }
    public String getNazwa() { return OWCA_NAZWA; }
}

File: projekt_1/Projekt_1/Projekt_1/Owca.cpp
#pragma once
#include "Owca.h"
#include <iostream>
using namespace std;

Owca::Owca(Swiat* swiat, int x, int y) : Zwierze(swiat, x, y, OWCA_SILA, OWCA_INIC, OWCA_PELNOLETNOSC) {}

bool Owca::Koliduj(Organizm* somsiad, bool czyAtakuje) {
	return false;
}

void Owca::dodajPolog() { this->polog = OWCA_POLOG; }
Organizm* Owca::dajPotomka(Wsp n_wsp) {
	Organizm* tmp = new Owca(swiat, n_wsp.x, n_wsp.y);
	swiat->DodajOrganizm(tmp);
	return tmp;
}
bool Owca::czyTenSamGatunek(Organizm* o) { if (o->getChar() == OWCA_CHAR) { return true; } else return false; }

char Owca::getChar() { return OWCA_CHAR; }
string Owca::getNazwa() { return OWCA_NAZWA; }
